specifyweb/frontend/js_src/lib/components/Formatters/Definitions.tsx

```diff
@@ -0,0 +1,241 @@
+    },
+    [formatter, setFormatter]
+  );
+
+  const hasCondition = formatter.definition.conditionField !== undefined;
+  const trimmedFields = hasCondition
+    ? formatter.definition.fields
+    : formatter.definition.fields.slice(0, 1);
+
+  const table = formatter.table;
+
+  const handleChanged = (
+    field: Formatter['definition']['fields'][number],
+    index: number
+  ): void =>
+    handleChange(replaceItem(formatter.definition.fields, index, field));
+
+  const needFormatter = formatter.definition.fields.length === 0;
+  React.useEffect(() => {
+    if (needFormatter) {
+      handleChange([
+        ...formatter.definition.fields,
+        {
+          value: undefined,
+          fields: [],
+        },
+      ]);
+    }
+  }, [needFormatter, handleChange, formatter.definition.fields]);
+
+  return table === undefined ? null : (
+    <div className="flex flex-col gap-4 divide-y divide-gray-500 [&>*]:pt-4">
+      {trimmedFields.map(({ value, fields }, index) => (
+        <ConditionalFormatter
+          fields={fields}
+          formatter={formatter}
+          hasCondition={hasCondition}
+          index={index}
+          key={index}
+          table={table}
+          trimmedFieldsLength={trimmedFields.length}
+          value={value}
+          onChange={handleChange}
+          onChanged={handleChanged}
+        />
+      ))}
+      {!isReadOnly && hasCondition ? (
+        <div>
+          <Button.Info
+            title={
+              hasCondition ? undefined : resourcesText.addConditionFieldFirst()
+            }
+            onClick={(): void =>
+              handleChange([
+                ...formatter.definition.fields,
+                {
+                  value: undefined,
+                  fields: [],
+                },
+              ])
+            }
+          >
+            {resourcesText.addDefinition()}
+          </Button.Info>
+        </div>
+      ) : null}
+    </div>
+  );
+}
+
+function ConditionalFormatter({
+  value,
+  fields,
+  hasCondition,
+  index,
+  onChanged: handleChanged,
+  trimmedFieldsLength,
+  formatter,
+  onChange: handleChange,
+  table,
+}: {
+  readonly value: LocalizedString | undefined;
+  readonly fields: Formatter['definition']['fields'][number]['fields'];
+  readonly hasCondition: boolean;
+  readonly index: number;
+  readonly onChanged: (
+    field: Formatter['definition']['fields'][number],
+    index: number
+  ) => void;
+  readonly trimmedFieldsLength: number;
+  readonly formatter: GetSet<Formatter>[0];
+  readonly onChange: (fields: Formatter['definition']['fields']) => void;
+  readonly table: SpecifyTable;
+}): JSX.Element {
+  const isReadOnly = React.useContext(ReadOnlyContext);
+  const [isExpanded, _, __, handleToggle] = useBooleanState(false);
+  return (
+    <div
+      className={`flex ${
+        isExpanded || !hasCondition ? 'flex-col' : 'items-center'
+      } gap-2`}
+      key={index}
+    >
+      {hasCondition && (
+        <Label.Block>
```

Comment:
The button is not supposed to be inside the label - labe should only contain the input field